HW1/Main.hs

```diff
@@ -0,0 +1,194 @@
+module Main where
+
+import Control.Monad (unless)
+import Text.Printf (printf)
+
+short :: [a] -> Bool
+short [] = True
+short [_] = True
+short [_, _] = True
+short _ = False
+
+lovely :: [Int] -> Bool
+lovely [] = True
+lovely [_] = True
+lovely [_, _] = True
+lovely (x : y : z : _) = z == 14
+
+rightTriangles :: [(Int, Int, Int)]
+rightTriangles = [(x, y, z) | z <- [5 ..], y <- [3 .. z], x <- [2 .. y], x * x + y * y == z * z]
+
+fizzBuzz :: [String]
+fizzBuzz =
+  [ elem | x <- [1 ..], let elem
+                              | x `mod` 3 == 0 && x `mod` 5 == 0 = "FizzBuzz"
+                              | x `mod` 5 == 0 = "Buzz"
+                              | x `mod` 3 == 0 = "Fizz"
+                              | otherwise = show x
+  ]
+
+ageOn :: String -> Float -> Float
+ageOn planet ageInSeconds
+  | planet == "Earth" = ageInSeconds / 31557600
+  | planet == "Mercury" = ageInSeconds / (0.2408467 * 31557600)
+  | planet == "Venus" = ageInSeconds / (0.61519726 * 31557600)
+  | planet == "Mars" = ageInSeconds / (1.8808158 * 31557600)
+  | planet == "Jupiter" = ageInSeconds / (11.862615 * 31557600)
+  | planet == "Saturn" = ageInSeconds / (29.447498 * 31557600)
+  | planet == "Uranus" = ageInSeconds / (84.016846 * 31557600)
+  | planet == "Neptune" = ageInSeconds / (164.79132 * 31557600)
+  | otherwise = error "planet undifined"
```

Comment:
Avoid repeating the same magic constant by binding a name to it.

Reply:
I have committed the changed code